### PR TITLE
[HS-35] Google Sign in - Success flow

### PR DESCRIPTION
**Pull request by asoi1990**

### CODE REVIEW

File: lib/features/sign_up/view_model/sign_up_bloc.dart [hunk 1]
```diff
@@ -27,5 +34,17 @@ class SignUpBloc extends Bloc<SignUpEvent, SignUpState> {
...
+
+    on<SignUpWithGoolge>((event, emit) async {
+      try {
+        UserDetail? user = await _authenticationService.signUpWithGoogle();
+        if (user != null) {
+          return emit(const SignUpSuccess());
+        }
+      } on PlatformException {
+        // TODO: SHOW TOAST
+        emit(SignUpFailed(SignUpStatusMessage.authenticationFaildMessage));
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/sign_up_bloc.dart:46]: belong to HS-50, but should not show hard coded text this way. Please use l10n

File: lib/models/authentication/authentication_service.dart [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  Future<UserDetail> getCurrentUser() async {
+    final User? firebaseUser = _firebaseAuth.currentUser;
+    if (firebaseUser == null) {
+      throw FirebaseAuthException(
+          code: "USER_NOT_FOUND", message: "User not found");
+    }
+    return UserDetail(
+        uid: firebaseUser.uid,
+        phone: firebaseUser.phoneNumber,
+        email: firebaseUser.displayName);
```
Comment (hellomentorplus) [inline lib/models/authentication/authentication_service.dart:59]: need email

**Review by hellomentorplus — CHANGES_REQUESTED** [review]

File: lib/features/sign_up/view_model/sign_up_event.dart [hunk 3]
```diff
@@ -3,14 +3,28 @@ part of 'sign_up_bloc.dart';
...
-  @override
-  List<Object> get props => [];
+  // @override
+  // List<Object> get props => [];
 }
 
 class CheckFirstLaunchSignUp extends SignUpEvent {
   const CheckFirstLaunchSignUp();
+  @override
+  // TODO: implement props
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/sign_up_event.dart:13]: if it's not your TODO, just delete it

File: lib/models/authentication/authentication_service.dart [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      //       code: "AUTH_CANCELED", message: "Account already create");
+      // }
+    } on FirebaseAuthException catch (e) {
+      throw {e.code, e.message};
+    }
+  }
+
+  Future<UserDetail> getCurrentUser() async {
+    final User? firebaseUser = _firebaseAuth.currentUser;
+    if (firebaseUser == null) {
```
Comment (hellomentorplus) [inline lib/models/authentication/authentication_service.dart:53]: define our own Exception and throw our own exception here

**Review by hellomentorplus — CHANGES_REQUESTED** [review]

File: lib/features/sign_up/view_model/sign_up_bloc.dart [hunk 5]
```diff
@@ -23,5 +32,17 @@ class SignUpBloc extends Bloc<SignUpEvent, SignUpState> {
       prefs.setBool(isFirstLaunchConst, false);
       emit(const FirstLaunchScreen(false));
     });
+
+    on<SignUpWithGoolge>((event, emit) async {
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/sign_up_bloc.dart:36]: SignUpWithGoogle

File: lib/features/sign_up/view_model/sign_up_state.dart [hunk 6]
```diff
@@ -20,3 +17,16 @@ class FirstLaunchScreen extends SignUpState {
...
 }
+
+class SignUpSuccess extends SignUpState {
+  const SignUpSuccess();
+  @override
+  List<Object?> get props => [];
+}
+
+class SignUpFailed extends SignUpState {
+  String errorCode;
```
Comment (hellomentorplus) [inline lib/features/sign_up/view_model/sign_up_state.dart:28]: all fields in State class need to be `final`

File: lib/models/authentication/authentication_service.dart [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+class UserDetail {
+  final String? uid;
+  final String? phone;
+  final String? email;
+
+  UserDetail({required this.uid, this.phone, this.email});
+}
+
+class SignUpStatusMessage {
+  static const alreadyHaveAccount = SignUpMessageType.AlreadyHaveAccount;
```
Comment (hellomentorplus) [inline lib/models/authentication/authentication_service.dart:69]: do not translate enum to string.

File: lib/l10n/intl_en.arb [hunk 8]
```diff
@@ -23,5 +23,7 @@
   "continueBtn":"Continue",
   "cancelBtn":"Cancel",
   "homePageViewTitle":"Home Page",
-  "linearProgressIndicator":"Linear progress indicator"
+  "linearProgressIndicator":"Linear progress indicator",
+  "AlreadyHaveAccount":"Already have account",
```
Comment (hellomentorplus) [inline lib/l10n/intl_en.arb:27]: if it's not required in story, do not create new strings

**Review by hellomentorplus — CHANGES_REQUESTED** [review]

**Review by hellomentorplus — APPROVED** [review]

**Review by suesitran — APPROVED** [review]

**Pull request merged**